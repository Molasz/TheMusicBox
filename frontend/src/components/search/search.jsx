import React from 'react';
import { connect } from 'react-redux';

import './search.scss';

import ListItem from './listItem/listItem';

function Search({ band }) {
  const list = band ? (
    band.map((element, i) => {
      return (
        <div className='list__item' key={i}>
          <ListItem data={element} />
        </div>
      );
    })
  ) : (
    <h1>Loading...</h1>
  );

  return <section className='list'>{list}</section>;
}

function mapStateToProps(state) {
  return { band: state.bandReducer.search };
}

export default connect(mapStateToProps, null)(Search);
export { Search };
