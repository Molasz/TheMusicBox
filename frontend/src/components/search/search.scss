.list {
  width: 80%;
  margin: 50px auto;

  display: flex;
  flex-direction: column;

  color: white;
  min-width: 500px;
  .list__item {
    display: flex;
    height: 150px;
  }
}
