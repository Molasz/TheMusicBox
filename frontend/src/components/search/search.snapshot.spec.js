import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { Search } from './search';

describe('Search snapshot', () => {
  it('Should match with band data', () => {
    const band = [
      {
        name: 'name',
        city: 'city',
        country: 'country',
        logo: 'logo.com',
        bio: 'blablabla',
        tags: ['punk', 'rock', 'heavy'],
        _id: 0
      }
    ];
    const tree = renderer.create(
      <BrowserRouter>
        <Search band={band} />
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
