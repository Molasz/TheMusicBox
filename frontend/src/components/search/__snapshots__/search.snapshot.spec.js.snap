// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot Should match with data 1`] = `
<section
  className="list"
>
  <div
    className="list__item"
  >
    0
  </div>
  <div
    className="list__item"
  >
    0
  </div>
</section>
`;

exports[`Search snapshot Should match without data 1`] = `
<section
  className="list"
>
  <h1>
    Loading...
  </h1>
</section>
`;
